server-connection/src/main/java/org/sonarsource/sonarlint/core/serverconnection/ServerPathProvider.java

```diff
@@ -33,23 +34,35 @@ private ServerPathProvider() {
 
   private static final String MIN_SQ_VERSION = "9.7";
 
-  public static String getServerUrlForTokenGeneration(EndpointParams endpoint, HttpClient client,
-    int port, String ideName) throws ExecutionException, InterruptedException {
+  public static CompletableFuture<String> getServerUrlForTokenGeneration(EndpointParams endpoint, HttpClient client,
+    int port, String ideName) {
     var serverApi = new ServerApi(endpoint, client);
-    var systemInfo = serverApi.system().getStatus().get();
-    return buildServerPath(endpoint.getBaseUrl(), systemInfo.getVersion(), port, ideName, endpoint.isSonarCloud());
+    CompletableFuture<ServerInfo> systemInfoFuture;
+    systemInfoFuture = serverApi.system().getStatus();
+    return systemInfoFuture.thenApply(systemInfo ->
+      buildServerPath(endpoint.getBaseUrl(), systemInfo.getVersion(), port, ideName, endpoint.isSonarCloud()));
   }
 
+
+
   static String buildServerPath(String baseUrl, String serverVersionStr, int port, String ideName, boolean isSonarCloud) {
     var minVersion = Version.create(MIN_SQ_VERSION);
     var serverVersion = Version.create(serverVersionStr);
     var path = new StringBuilder(baseUrl);
+    var portParameter = getPortParameter(port);
     if (isSonarCloud || !serverVersion.satisfiesMinRequirement(minVersion)) {
       path.append("/account/security");
     } else {
-      path.append("/sonarlint/auth").append("?port=").append(port).append("&ideName=").append(UrlUtils.urlEncode(ideName));
+      path.append("/sonarlint/auth").append("?ideName=").append(UrlUtils.urlEncode(ideName)).append(portParameter);
     }
     return path.toString();
   }
 
+  private static String getPortParameter(int port) {
+    if (port == -1) {
```

Comment:
As I said during a discussion, I find it a bit strange to "encode" the fact that there is no port with a `-1`. By looking at the method API it's not obvious compared to a Nullable or Optional parameter (or even 2 different methods, even if the -1 is then used internally). And as it is not documented it could be hard to guess